processing/src/main/java/io/druid/segment/data/VSizeLongSerde.java

```diff
@@ -0,0 +1,633 @@
+  {
+    void write(long value) throws IOException;
+  }
+
+  private static final class Size1Ser implements LongSerializer
+  {
+    OutputStream output = null;
+    ByteBuffer buffer;
+    byte curByte = 0;
+    int count = 0;
+
+    public Size1Ser(OutputStream output)
+    {
+      this.output = output;
+      this.buffer = ByteBuffer.allocate(1);
+    }
+
+    public Size1Ser(ByteBuffer buffer, int offset)
+    {
+      this.buffer = buffer;
+      this.buffer.position(offset);
+    }
+
+    @Override
+    public void write(long value) throws IOException
+    {
+      if (count == 8) {
+        buffer.put(curByte);
+        count = 0;
+        if (!buffer.hasRemaining() && output != null) {
+          output.write(buffer.array());
+          buffer.rewind();
+        }
+      }
+      curByte = (byte) ((curByte << 1) | (value & 1));
+      count++;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+      buffer.put((byte) (curByte << (8 - count)));
+      if (output != null) {
+        output.write(buffer.array());
+        output.write(EMPTY);
+        output.flush();
+      } else {
+        buffer.putInt(0);
+      }
+    }
+  }
+
+  private static final class Size2Ser implements LongSerializer
+  {
+    OutputStream output = null;
+    ByteBuffer buffer;
+    byte curByte = 0;
+    int count = 0;
+
+    public Size2Ser(OutputStream output)
+    {
+      this.output = output;
+      this.buffer = ByteBuffer.allocate(1);
+    }
+
+    public Size2Ser(ByteBuffer buffer, int offset)
+    {
+      this.buffer = buffer;
+      this.buffer.position(offset);
+    }
+
+    @Override
+    public void write(long value) throws IOException
+    {
+      if (count == 8) {
+        buffer.put(curByte);
+        count = 0;
+        if (!buffer.hasRemaining() && output != null) {
+          output.write(buffer.array());
+          buffer.rewind();
+        }
+      }
+      curByte = (byte) ((curByte << 2) | (value & 3));
+      count += 2;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+      buffer.put((byte) (curByte << (8 - count)));
+      if (output != null) {
+        output.write(buffer.array());
+        output.write(EMPTY);
+        output.flush();
+      } else {
+        buffer.putInt(0);
+      }
+    }
+  }
+
+  private static final class Mult4Ser implements LongSerializer
+  {
+
+    OutputStream output = null;
+    ByteBuffer buffer;
+    int numBytes;
+    byte curByte = 0;
+    boolean first = true;
+
+    public Mult4Ser(OutputStream output, int numBytes)
+    {
+      this.output = output;
+      this.buffer = ByteBuffer.allocate(numBytes * 2 + 1);
+      this.numBytes = numBytes;
+    }
+
+    public Mult4Ser(ByteBuffer buffer, int offset, int numBytes)
+    {
+      this.buffer = buffer;
+      this.buffer.position(offset);
+      this.numBytes = numBytes;
+    }
+
+    @Override
+    public void write(long value) throws IOException
+    {
+      int shift = 0;
```

Comment:
is `shift` variable needed? Seems like it gets set to 4 once before it's used and never changes after